docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,285 @@
+6. The scope of the data that the team admin can access depends on how the Redis cluster is deployed. If there is a single Redis cluster for all the namespaces, the team admin can read data from other namespaces. If there is a separate Redis cluster for each namespace, the team admin can only access data in their own namespace.
+
+##### Cache Partition
+1. If each namespace has its own Redis instance, there is no need for cache partitioning, because the data is already separated by namespace.
+2. If there is a single Redis cluster for the whole cluster, cache partitioning is necessary, because different teams might have data with the same key. In this case, we can use the namespace as a prefix for the key to avoid conflict.
+
+##### Redis Deployment and Ratify Deployment
+The deployment strategy affects the Cache Isolation, as shown above.
+
+Ratify can be deployed either per namespace or per cluster. Redis can also be deployed either per namespace or per cluster. The table below shows the different deployment combinations and their approaches to cache isolation.
+
+|  | Ratify per Namespace | Ratify per Cluster |
+|--|--|--|
+| Redis per Namespace | Data Isolated | It requires Ratify route requests to Redis instances in different namespaces |
+| Redis per Cluster | Prefix keys and restrict access via [ACL](https://redis.io/docs/management/security/acl/) | Prefix keys and restrict access via [ACL](https://redis.io/docs/management/security/acl/) |
+
+#### Log
+Ratify saves logs to the standard output and then handled by container runtime. Whatever path does container runtime save logs, those logs may include data from different teams. This situation mirrors the challenges observed with cache isolation.
+
+##### User Scenario
+1. Cluster admin could inspect logs from all namespaces via `kubectl logs` command.
+2. Team admin could inspect logs from its own namespace  via `kubectl logs` command but has no access to logs from other teams.
+3. If users have set up log pipeline on Ratify such as ELK, they can forbid `kubectl logs` and enforce log inspection through Kibana.
+
+##### Isolation
+1. If Ratify is deployed per namespace, then logs are isolated by design.
+2. If Ratify is deployed once in cluster, there should be some mechanism to partition logs.
+    * One solution is adding a namespace field to all Ratify logs. Although this option cannot avoid cross-tenant access without disabling `kubectl logs`, it can be used to filter logs in Kibana.
+    * The other solution is to let Ratify partition logs by path. For example, logs from namespaceA are saved to `/var/log/namespaceA` and logs from namespaceB are saved to `/var/log/namespaceB`. This solution can avoid cross-tenant access without disabling `kubectl logs`. However, it requires Ratify to configure logrus output paths for each team explicitly. However, Ratify has to dynamically configure paths for all namespaces, including creating and deleting directories.
+
+#### File system/PVC
```

Comment:
We could add priority to the areas that requires multi-tenancy support. The FileSystem is probably higher pri item and should be in listed towards the top of the doc.

Reply:
My initial plan was to have another doc for implementation details once we decide the model.

Reply:
here is a brief work break down: https://hackmd.io/@H7a8_rG4SuaKwzu4NLT-9Q/By-jVTBST